Bottom: fb3f150e6146757e6bdb884182c4b19457bd6b6b
Top:    456260422aff51feca00c1ee2db83c3d63b6e111
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-16 15:17:13 -0700

Refresh of rs-nodelay

---

diff --git a/examples/rstream.c b/examples/rstream.c
index decb204..8d5a22d 100644
--- a/examples/rstream.c
+++ b/examples/rstream.c
@@ -57,7 +57,6 @@ static int use_rs = 1;
 static int use_async;
 static int verify;
 static int flags = MSG_DONTWAIT;
-static int no_delay = 1;
 static int custom;
 static int iterations = 1;
 static int transfer_size = 1000;
@@ -324,10 +323,8 @@ static void set_options(int rs)
 			break;
 	}
 
-	if (no_delay) {
-		rs_setsockopt(rs, IPPROTO_TCP, TCP_NODELAY,
-			      (void *) &no_delay, sizeof(no_delay));
-	}
+	rs_setsockopt(rs, IPPROTO_TCP, TCP_NODELAY,
+		      (void *) &no_delay, sizeof(no_delay));
 
 	if (flags & MSG_DONTWAIT) {
 		rs_fcntl(rs, F_SETFL, O_NONBLOCK);
@@ -488,11 +485,9 @@ static int set_test_opt(char *optarg)
 			break;
 		case 'b':
 			flags &= ~MSG_DONTWAIT;
-			no_delay = 0;
 			break;
 		case 'n':
 			flags |= MSG_DONTWAIT;
-			no_delay = 1;
 			break;
 		case 'v':
 			verify = 1;
@@ -507,10 +502,8 @@ static int set_test_opt(char *optarg)
 			use_async = 1;
 		} else if (!strncasecmp("block", optarg, 5)) {
 			flags &= ~MSG_DONTWAIT;
-			no_delay = 0;
 		} else if (!strncasecmp("nonblock", optarg, 8)) {
 			flags |= MSG_DONTWAIT;
-			no_delay = 1;
 		} else if (!strncasecmp("verify", optarg, 6)) {
 			verify = 1;
 		} else {
